pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +540,90 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+def _find_solve_with_eye(node) -> bool:
+    """
+    The result of solve(A, b) is the solution x to the linear equation Ax = b. If b is an identity matrix (Eye), x is simply inv(A).
+    Here, we are just recognising whether the solve operation returns an inverse or not; not replacing it because solve is mathematically more stable than inv.
+    """
+    valid_solves = (Solve, SolveTriangular)
+    # First, we verify whether we have a valid solve op
+    if not (
+        isinstance(node.op, Blockwise) and isinstance(node.op.core_op, valid_solves)
+    ):
+        return False
+    # If the current op is solve, we check for b. If b is an identity matrix (Eye), we can return True
+    solve_inputs = node.inputs
+    eye_input = solve_inputs[1].owner
```

Comment:
Doesn't make sense to call this object _input
```suggestion
    eye_node = solve_inputs[1].owner
```